worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -1246,6 +1215,28 @@ class PushDataHandler(val workerSource: WorkerSource) extends BaseMessageHandler
     }
   }
 
+  def writeDataWithExceptionHandling(
+      fileWriter: FileWriter,
+      body: ByteBuf,
+      shuffleKey: String): Future[Unit] = {
+    Future {
+      try {
+        fileWriter.write(body)
+      } catch {
+        case e: AlreadyClosedException =>
```

Comment:
The original logic has another case to catch the general `Exception`, but here we only catch for `AlreadyClosedException`. Why the case for `Exception` is dropped here or am I missing something?

Reply:
Yes, but all they do with that is log the exception. In the new code, I am throwing it anyways, which will get propagated/logged anyways. It goes via `callbackWithTimer.onFailure(e)` to client, which will log the error.

Reply:
I see, thanks!